### PR TITLE
Adding a monocular client as a package

### PR DESCRIPTION
**Pull request by mattfarina**
Includes client creation and search

Relates to #6186

### CODE REVIEW

File: pkg/monocular/search.go
Comment (thomastaylor312) [inline pkg/monocular/search.go]: Do these structs exist in monocular and could we use them if so?

File: pkg/monocular/search.go
Comment (thomastaylor312) [inline pkg/monocular/search.go]: Consider making this part of the path a constant

**Review by thomastaylor312 — COMMENTED** [review]
This is looking pretty good.

Just wanted to have one discussion around whether or not this should be in `internal` or not. I could see this code being useful to others, but worry about exporting it and supporting it still. A full monocular client feels like something that should belong in monocular and be imported

File: pkg/monocular/search.go
Comment (mattfarina) [reply pkg/monocular/search.go]: They do. But, the package imports `k8s.io/helm/pkg/proto/hapi/chart`. I'd like to avoid that package import from helm v2 to helm v3. Any ideas other than copying?

**Comment by bacongobbler** [comment]
I noticved a lot of code in here shares similarities with `pkg/getter/httpgetter.go`. Would it be worth conslidating the HTTP client creation code? For example, the `tlsutil` setup as well as the proxy setup available in httpgetter may be worthwhile to consider.

**Comment by bacongobbler** [comment]
https://github.com/helm/helm/blob/affd77558ffd343716036c01a075ed16540c5aaa/pkg/getter/httpgetter.go#L79-L100

**Comment by bacongobbler** [comment]
I even notice that the same HTTP user agent introduce here is commented out in httpgetter.go... Perhaps we can just uncomment that. Not sure why it's unavailable :thinking: 

https://github.com/helm/helm/blob/affd77558ffd343716036c01a075ed16540c5aaa/pkg/getter/httpgetter.go#L49

**Comment by mattfarina** [comment]
@bacongobbler The user agent used to be set for the httpgetter. When we used to pull chart stats we could see which versions of Helm were calling for it. On master this is still happening. Somewhere in the v3 refactor something changed.

The httpgetter is coupled to the getter system. Is it worth trying to make the two work using the same code? Should the getter system (or just the httpgetter) be used for the basic http request?

**Comment by bacongobbler** [comment]
It might be worth re-using the getter system for monocular. While monocular does not interface directly with a basic HTTP filestore similar to the chart repository API (s3, azure, chartmuseum, gcs), the function signatures for `.Search(term)` and `.Get(url)` seem similar enough in design WRT HTTP client setup. I'll experiment with this today but I wouldn't block this from going through.

The user agent string on the httpgetter should probably be restored. Consumers can overwrite the user agent string if they choose. I'll file a separate PR for that.

**Comment by mattfarina** [comment]
@thomastaylor312 I made the tweaks you suggested. In addition to that I added a user agent function to the version package since we will need that in a couple other places. This is after chatting with @bacongobbler who is specifying the user agent in another PR.

**Comment by bacongobbler** [comment]
#6197 has already been merged, so if you want to update that part of the codebase to use the `.UserAgent` function, that would be dandy. :)

**Comment by mattfarina** [comment]
@bacongobbler done!

**Review by bacongobbler — APPROVED** [review]

File: pkg/monocular/search.go
Comment (thomastaylor312) [reply pkg/monocular/search.go]: Nope, that makes complete sense

**Review by thomastaylor312 — APPROVED** [review]

**Comment by waveywaves** [comment]
There seems to be conflict merges on `internal/version/version.go`

**Comment by mattfarina** [comment]
@waveywaves I knew this was coming. @thomastaylor312 warned me before merging his change that caused this. A rebase with the most minor of changes (i.e.i removing an import) was the only thing needed to correct this.

**Pull request merged**
